File: FootyFocus-Frontend/src/app/models/PlayerInfo.ts
export interface PlayerInfo{
  
  //player information
  firstname: string;
  lastname: string;
  age: number;
  height: string;
  weight: string;
  injured: boolean;
  photo: string;
  birthDate: string;
  birthPlace: string;
  position: string;
  captain: boolean;

  //player team information
  teamName: string;
  teamLogo: string;
  
  //player statistics
  //shots and saves
  goals: number;
  assists: number;
  totalShots: number;
  shotsOnTarget: number;
  saves: number;

  //passes
  totalPasses: number;
  keyPasses: number;
  passAccuracy: number;
  
  //defence
  totalTackles: number;
  blocks: number;
  interceptions: number;
  totalDuels: number;
  duelsWon: number;

  //dribbles
  dribbleAttempts: number;
  successfulDribbles: number;

  //fouls
  foulsDrawn: number;
  foulsCommited: number;

  //cards
  yellow: number;
  yellowRedCard: number;
  red: number;

  // penalties
  penaltiesWon: number;
  penaltiesCommited: number;
  penaltiesScored: number;
  penaltiesMissed: number;
  penaltiesSaved: number;
}

File: FootyFocus-Backend/src/main/java/com/example/FootyFocus/entity/PlayerInfo.java
package com.example.FootyFocus.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class PlayerInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String firstname;
    private String lastname;
    private int age;
    private String height;
    private String weight;
    private boolean injured;
    private String photo;

    private String birthDate;
    private String birthPlace;

    private String teamName;
    private String teamLogo;

    private int appearances;
    private String position;
    //private int rating;
    private boolean captain;

    private int totalShots;
    private int shotsOnTarget;

    private int goals;
    private int goalsConceded;
    private int assists;
    private int saves;  // Nullable field

    private int totalPasses;
    private int keyPasses;
    private int passAccuracy;

    private int totalTackles;
    private int blocks;
    private int interceptions;

    private int totalDuels;
    private int duelsWon;

    private int dribbleAttempts;
    private int successfulDribbles;

    private int foulsDrawn;
    private int foulsCommitted;

    private int yellowCards;
    private int yellowRedCards;
    private int redCards;

    private int penaltiesWon;
    private int penaltiesCommitted;
    private int penaltiesScored;
    private int penaltiesMissed;
    private int penaltiesSaved;

}
